pychromecast/__init__.py

```diff
@@ -194,6 +197,13 @@ def __init__(self, host, port=None, device=None, **kwargs):
         if blocking:
             self.socket_client.start()
 
+        # Set to True by default if cast_type == CAST_TYPE_GROUP
+        self.dynamic_host = True if self.cast_type == CAST_TYPE_GROUP \
+            else False
+        self.dynamic_host = kwargs.pop('dynamic_host', self.dynamic_host)
+        if self.dynamic_host:
+            self.start_dynamic_host_tracking()
```

Comment:
Yeah, as I understand, CPython doesn't provide any guarantee objects will necessarily be cleaned up in a timely manner. They should at least all be cleaned up with the interpreter is shutting down, though.